File: Lab3/app/src/main/java/com/example/pamo/lab3/Gender.java
package com.example.pamo.lab3;

import java.util.stream.Stream;

public enum Gender {
    MALE(R.id.maleRadio),
    FEMALE(R.id.femaleRadio);

    private final Integer radioId;

    Gender(Integer radioId) {
        this.radioId = radioId;
    }

    public static Gender getGenderById(Integer id) {
        return Stream.of(Gender.values()).filter(v -> v.radioId.equals(id)).findFirst().orElse(null);
    }

    public Integer getRadioId() {
        return radioId;
    }
}

File: Lab4/app/src/main/java/com/example/pamo/lab4/Gender.kt
package com.example.pamo.lab4

enum class Gender(val radioId: Int) {
    MALE(R.id.maleRadio), FEMALE(R.id.femaleRadio);

    companion object {
        fun getGenderById(id: Int): Gender {
            return values().first { v: Gender -> v.radioId == id }
        }
    }
}